pylint/pyreverse/inspector.py

```diff
@@ -313,6 +325,55 @@ def _imported_module(
                 mod_paths.append(mod_path)
 
 
+class AssociationHandlerInterface(ABC):
+    @abstractmethod
+    def set_next(
+        self, handler: AssociationHandlerInterface
+    ) -> AssociationHandlerInterface:
+        pass
+
+    @abstractmethod
+    def handle(self, node: nodes.AssignAttr, parent: nodes.ClassDef) -> None:
+        pass
+
+
+class AbstractAssociationHandler(AssociationHandlerInterface):
+    """
+    The default chaining behavior can be implemented inside a base handler
+    class.
```

Comment:
I think it would be great to add some more explanation here why we need this chaining behaviour. It took me a while to see why you implemented it like this and don't just loop over all possible handlers.
But I think it is a good solution for the problem and allows for extension later on if we want to handle other association types better as well.

Reply:
Updated docs :+1: